### ISSUE TITLE
Dependency failing: com.google.android.gms:play-services-clearcut:11.0.4 -> com.google.android.gms:play-services-tasks@[   11.0.4], but play-services-tasks version was 17.0.0

### ISSUE THREAD

**Issue by rovada3909** [andrehtissot/cordova-plugin-fcm-with-dependecy-updated#23]
Hi,
   I am developing app with Ionic Cordova. I have following dependency in app gradle 

```gradle

dependencies {
    implementation fileTree(dir: 'libs', include: '*.jar')
    // SUB-PROJECT DEPENDENCIES START
    implementation(project(path: ":CordovaLib"))
    implementation "com.google.android.gms:play-services-base:11.0.4"
    implementation "com.google.android.gms:play-services-ads:11.0.4"
    implementation "com.google.firebase:firebase-messaging:19.0.0"
    implementation "com.android.support:support-annotations:27.+"
    // SUB-PROJECT DEPENDENCIES END 
 }

```

Ionic CLI 5.4.16

### Plugins

```
cordova-admob-sdk 0.24.1 "AdMob SDK"
cordova-plugin-admob-free 0.27.0 "Cordova AdMob Plugin"
cordova-plugin-device 2.0.2 "Device"
cordova-plugin-fcm-with-dependecy-updated 4.4.1 "Cordova FCM Push Plugin"
cordova-plugin-ionic-keyboard 2.2.0 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 4.1.3 "cordova-plugin-ionic-webview"
cordova-plugin-nativestorage 2.3.2 "NativeStorage"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-statusbar 2.4.2 "StatusBar"
cordova-plugin-whitelist 1.3.3 "Whitelist"
cordova-promise-polyfill 0.0.2 "cordova-promise-polyfill"
cordova-support-android-plugin 1.0.1 "cordova-support-android-plugin"
cordova-support-google-services 1.4.0 "cordova-support-google-services"

```

app is building fine with above dependency. But problem arrives as soon as install following plugin

`ionic cordova plugin add cordova-plugin-fcm-with-dependecy-updated`

I couldn't find out exact problem here. Seems like play services dependency version mismatch.

Please  help me. Thanks.

Here the error log.

```console
> Configure project :app
FCM PLUGIN GRADLE TOOLS VERSION: 3.5.3
> Task :app:preBuild UP-TO-DATE
> Task :CordovaLib:preBuild UP-TO-DATE
> Task :CordovaLib:preDebugBuild UP-TO-DATE
> Task :CordovaLib:checkDebugManifest UP-TO-DATE
> Task :CordovaLib:processDebugManifest UP-TO-DATE


FAILURE: Build failed with an exception.

* What went wrong:
Failed to capture fingerprint of input files for task ':app:preDebugBuild' property 'compileManifests' during up-to-date check.
> In project 'app' a resolved Google Play services library dependency depends on another at an exact 
version (e.g. "[11.0.
  4]", but isn't being resolved to that version. Behavior exhibited by the library will be unknown.  

  Dependency failing: com.google.android.gms:play-services-clearcut:11.0.4 -> com.google.android.gms:play-services-tasks@[
  11.0.4], but play-services-tasks version was 17.0.0.

  The following dependencies are project dependencies that are direct or have transitive dependencies that lead to the art
  ifact with the issue.
  -- Project 'app' depends onto com.google.android.gms:play-services-ads@11.0.4
  -- Project 'app' depends onto com.google.android.gms:play-services-clearcut@11.0.4
  -- Project 'app' depends onto com.google.android.gms:play-services-tasks@17.0.0
  -- Project 'app' depends onto com.google.android.gms:play-services-base@17.0.0
  -- Project 'app' depends onto com.google.firebase:firebase-common@18.0.0
  -- Project 'app' depends onto com.google.android.gms:play-services-safetynet@11.0.4
  -- Project 'app' depends onto com.google.firebase:firebase-iid-interop@17.0.0
  -- Project 'app' depends onto com.google.firebase:firebase-iid@19.0.0
  -- Project 'app' depends onto com.google.firebase:firebase-messaging@19.0.0
  -- Project 'app' depends onto com.google.android.gms:play-services-base@11.0.4

  For extended debugging info execute Gradle from the command line with ./gradlew --info :app:assembleDebug to see the dep
  endency paths to the artifact. This error message came from the google-services Gradle plugin, report issues at https://
  github.com/google/play-services-plugins and disable by adding "googleServices { disableVersionCheck = false }" to your b
  uild.gradle file.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 15s
> Task :app:preDebugBuild FAILED
Dependency resolved to an incompatible version: Dependency(fromArtifactVersion=ArtifactVersion(groupId=com.google.android.gms, artifactId=play-services-clearcut, version=11.0.4), toArtifact=Artifact(groupId=com.google.android.gms, artifactId=play-services-tasks), toArtifactVersionString=[11.0.4])      
3 actionable tasks: 1 executed, 2 up-to-date
E:\New folder\ElixirGoldNew\platforms\android\gradlew: Command failed with exit code 1 Error output:
FAILURE: Build failed with an exception.

* What went wrong:
Failed to capture fingerprint of input files for task ':app:preDebugBuild' property 'compileManifests' during up-to-date check.
> In project 'app' a resolved Google Play services library dependency depends on another at an exact 
version (e.g. "[11.0.
  4]", but isn't being resolved to that version. Behavior exhibited by the library will be unknown.  

  Dependency failing: com.google.android.gms:play-services-clearcut:11.0.4 -> com.google.android.gms:play-services-tasks@[
  11.0.4], but play-services-tasks version was 17.0.0.

  The following dependencies are project dependencies that are direct or have transitive dependencies that lead to the art
  ifact with the issue.
  -- Project 'app' depends onto com.google.android.gms:play-services-ads@11.0.4
  -- Project 'app' depends onto com.google.android.gms:play-services-clearcut@11.0.4
  -- Project 'app' depends onto com.google.android.gms:play-services-tasks@17.0.0
  -- Project 'app' depends onto com.google.android.gms:play-services-base@17.0.0
  -- Project 'app' depends onto com.google.firebase:firebase-common@18.0.0
  -- Project 'app' depends onto com.google.android.gms:play-services-safetynet@11.0.4
  -- Project 'app' depends onto com.google.firebase:firebase-iid-interop@17.0.0
  -- Project 'app' depends onto com.google.firebase:firebase-iid@19.0.0
  -- Project 'app' depends onto com.google.firebase:firebase-messaging@19.0.0
  -- Project 'app' depends onto com.google.android.gms:play-services-base@11.0.4

  For extended debugging info execute Gradle from the command line with ./gradlew --info :app:assembleDebug to see the dep
  endency paths to the artifact. This error message came from the google-services Gradle plugin, report issues at https://
  github.com/google/play-services-plugins and disable by adding "googleServices { disableVersionCheck = false }" to your b
  uild.gradle file.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 15s
[ERROR] An error occurred while running subprocess cordova.

        cordova.cmd build android exited with exit code 1.
        Re-running this command with the --verbose flag may provide more information.

```

**Comment by andrehtissot** [comment]
I was able to reproduce this issue and I'm looking into it.

**Comment by andrehtissot** [comment]
You were right about the nature of the issue.

Unfortunately, the only way I was able to install was downgrading the firebase-message dependency to also `11.0.4`, which is not currently supported due to the implementation targeting androidx.

I was able to build:
- Using the last version non-androidx: `cordova-plugin-fcm-with-dependecy-updated@2.4.0`;
- Changing the files `platforms/android/cordova-plugin-fcm-with-dependecy-updated/exampleapp-FCMPlugin.gradle` and `platforms/android/project.properties`, where it is `10.+` to `11.0.4`, matching with the version of the, conflicting, `cordova-admob-sdk` plugin.

I would, in your case, try to get the `cordova-admob-sdk`'s dependencies upgraded, making it  compatible with newer versions of play services.
